hollow/src/main/java/com/netflix/hollow/api/producer/AbstractHollowProducer.java

```diff
@@ -372,6 +373,7 @@ long runCycle(
 
             // 3. Produce a new state if there's work to do
             if (writeEngine.hasChangedSinceLastCycle()) {
+                writeEngine.addHeaderTag(HollowStateEngine.HEADER_TAG_SCHEMA_HASH, new HollowSchemaHash(writeEngine.getSchemas()).getHash());
                 boolean schemaChangedFromPriorVersion = readStates.hasCurrent() &&
```

Comment:
Wondering if `schemaChangedFromPriorVersion` should also be computed by comparing the SchemaHash values to ensure there is a sync between the 2 header values - HEADER_TAG_SCHEMA_HASH and HEADER_TAG_SCHEMA_CHANGE.